src/Converter.hh

```diff
@@ -100,9 +100,11 @@ namespace sdf
 
     /// \brief Remove an element.
     /// \param[in] _elem The element that has the _removeElem child.
-    /// \param[in] _removeElem The element to remove.
+    /// \param[in] _removeElem The metadata about what to remove.
+    /// \param[in] _removeEmpty If true, only remove empty nodes.
```

Comment:
Can you document how this works for attributes and elements? From what I can understand, for attributes, it removes just the attribute if it's empty. For elements, it removes the element if it has an empty value and has no children, but it can have attributes.

Reply:
would it be better to change the behavior for removing elements if they also have no attributes? We aren't yet using that functionality, so I couldn't decide. For now, I'll just document it as you suggest

Reply:
updated documentation and renamed the parameter to `_removeOnlyEmpty` in https://github.com/ignitionrobotics/sdformat/pull/720/commits/e6eb29391e12f320619d2ef9d408b847603b2c47

Reply:
Is it possible to chain these? By that I mean, would it be possible to remove the attributes first and then remove the element? If so, then I'd say let's make it so that it removes elements only if they have no attributes. This would make it possible to remove just the deprecated attributes first and remove the element if it doesn't contain anything else. For example:
```xml
<test_elem attr1="" attr2="some_val"/>
<test_elem attr1="" />
```

If we deprecate the use of `//test_elem` and know that `//test_elem[@attr1=""]` is the default, we can do 

```xml
<remove_empty attribute='attr1'/>
<remove_empty element='test_elem'/>
```
and it will only delete the second `//test_elem` and we would be able to issue a warning about the first `//test_elem`

Reply:
sure, I'll rework the function logic accordingly

Reply:
done in https://github.com/ignitionrobotics/sdformat/pull/720/commits/9588e0f82f78fa2ddb57e2c8a54339ca1625b55c